google-cloud-examples/src/main/java/com/google/cloud/examples/deployment-manager/DeploymentManagerDemo.java

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 Google LLC. This software is provided as-is, without warranty or representation for any use or purpose. Your use of it is subject to your agreement with Google.
+ */
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.deploymentmanager.DeploymentManager;
+import com.google.api.services.deploymentmanager.model.*;
+
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class DeploymentManagerDemo {
+
+    private static String PROJECTID = "project-id";
+    private static String CONFIG_YAML = "/path/to/deployment-manager-config.yaml";
+
+    // authentication uses the GOOGLE_APPLICATION_CREDENTIALS environemnt variable.
+    // run `gcloud auth application-default login` to create/update the file locally
+
+    // in intellij you can set this in the runtime config next to the run button
+    // private static String GOOGLE_APPLICATION_CREDENTIALS="/path/to/application_default_credentials.json";
+
+    // javadoc reference for deployment manger is here
+    // https://developers.google.com/resources/api-libraries/documentation/deploymentmanager/v2/java/latest/
+    public static void main(String args[]) throws GeneralSecurityException, IOException {
+
+        System.out.println("GOOGLE_APPLICATION_CREDENTIALS: " + System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+
+        // best practice is to use application default credential
+        GoogleCredential credential = GoogleCredential.getApplicationDefault();
+
+        // to load from somewhere else use this
+        // GoogleCredential credential = GoogleCredential.fromStream(new FileInputStream(GOOGLE_APPLICATION_CREDENTIALS));
+
+
+        HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+
+        DeploymentManager dm2 = new DeploymentManager.Builder(httpTransport, jsonFactory, credential)
+                .setApplicationName("New_Application")
+                .build();
+
+        Deployment deploy = new Deployment();
+        // Optional  we want some labels: Map of labels; provided by the client when the resource is created or updated.
+        List<DeploymentLabelEntry> labels = new ArrayList<DeploymentLabelEntry>();
```

Comment:
As of Java 8 you no longer need to include the specialization of the generic on the RHS. 
`List<DeploymentLabelEntry> labels = new ArrayList<>();`